### PR TITLE
Full Safe Redesign

### PR DESCRIPTION
**Pull request by biafra23**
Handles #1802 (#1804, #1805, #1808)

Changes proposed in this pull request:
- Fonts
- Colors
- Illustrations
- Strings 
- Links

Dark:
Assets | Transactions | App Settings | Safe Settings
------- | ------------ | ------------- | --------------
image | image | image | image

Light: 
Assets | Transactions | App Settings | Safe Settings
------- | ------------ | ------------- | --------------
image | image | image |image

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -115,6 +115,11 @@ android {
     }
 
     sourceSets {
+        main {
```
Comment (elgatovital) [inline app/build.gradle:118]: I would use same syntax as with other params:
```
main.res.srcDirs += ['src/main/res', 'src/main/res/font']
```

File: app/src/main/res/drawable-night/ic_coin_placeholder.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        android:pathData="M28 14A14 14 0 0 1 0 14A14 14 0 0 1 28 14Z"
+        android:fillColor="#303033" />
+    <path
+        android:pathData="M16.998 6.99951C13.133 6.99951 9.99799 10.1345 9.99799 13.9995C9.99799 17.8655 13.133 20.9995 16.998 20.9995C20.864 20.9995 23.998 17.8655 23.998 13.9995C23.998 10.1345 20.864 6.99951 16.998 6.99951M16.998 8.99951C19.756 8.99951 21.998 11.2425 21.998 13.9995C21.998 16.7575 19.756 18.9995 16.998 18.9995C14.241 18.9995 11.998 16.7575 11.998 13.9995C11.998 11.2425 14.241 8.99951 16.998 8.99951"
+        android:fillType="evenOdd"
+        android:fillColor="#636669" />
+    <path
+        android:pathData="M8.80447 7.35826C5.95932 8.29475 4.00012 10.9554 4.00012 13.9961C4.00012 17.0369 5.95932 19.6975 8.80447 20.634C9.32907 20.8067 9.89432 20.5214 10.067 19.9968C10.2397 19.4722 9.95437 18.9069 9.42977 18.7343C7.39868 18.0657 6.00012 16.1664 6.00012 13.9961C6.00012 11.8258 7.39868 9.92654 9.42977 9.258C9.95437 9.08533 10.2397 8.52008 10.067 7.99548C9.89432 7.47088 9.32907 7.18559 8.80447 7.35826Z"
+        android:fillColor="#636669" />
+</vector>
```
Comment (elgatovital) [inline app/src/main/res/drawable-night/ic_coin_placeholder.xml:16]: new line at the end missing

File: app/src/main/res/drawable-night/ic_empty_collectibles.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    <path
+        android:pathData="M38 28C38 22.4772 33.5228 18 28 18C22.4772 18 18 22.4772 18 28C18 33.5228 22.4772 38 28 38C33.5228 38 38 33.5228 38 28ZM22 28C22 24.6863 24.6863 22 28 22C31.3137 22 34 24.6863 34 28C34 31.3137 31.3137 34 28 34C24.6863 34 22 31.3137 22 28Z"
+        android:fillType="evenOdd"
+        android:fillColor="#636669" />
+    <path
+        android:pathData="M60.2448 73.3369C60.2368 73.6615 60.1482 73.9909 59.965 74.2972L59.8708 74.4413L37.2112 104.165C36.4441 105.171 34.9609 105.213 34.1341 104.291L34.0301 104.165L11.442 74.5354C11.4053 74.4894 11.3706 74.4417 11.338 74.3927C11.0806 74.0106 10.9773 73.5875 11.0001 73.1793C11.0136 72.8806 11.0965 72.5791 11.2589 72.2957C11.3098 72.2071 11.3661 72.1225 11.4286 72.0425L20.6601 59.5697C21.003 59.1064 21.5271 58.8159 22.0958 58.7669L22.2677 58.7595H49.8887C50.5014 58.7595 51.0771 59.0402 51.4545 59.5153L51.5524 59.6496L59.9587 72.2547C60.1443 72.5634 60.2347 72.8931 60.2442 73.2172L60.2448 73.3369ZM54.2 75.2805H17.04L35.621 99.6539L54.2 75.2805ZM23.276 62.7589H48.818L54.502 71.2799H16.969L23.276 62.7589Z"
+        android:fillType="evenOdd"
+        android:fillColor="#12FF80" />
+</vector>
```
Comment (elgatovital) [inline app/src/main/res/drawable-night/ic_empty_collectibles.xml:25]: image
Make sure to enable `Ensure every saved file ends with a line break` option

**Comment by elgatovital** [comment]
Separators for items in Help Center are missing, link icon color has to be adjusted for addresses.
In dark mode some separators are black instead of grey: Settings/Appearance, Tx Details screens

File: app/build.gradle (hunk 1 above)
Comment (biafra23) [reply app/build.gradle:118]: Sure. I think this was inserted by some tool. I don't remember adding it.

File: app/src/main/res/drawable-night/ic_empty_collectibles.xml (hunk 3 above)
Comment (biafra23) [reply app/src/main/res/drawable-night/ic_empty_collectibles.xml:25]: I have it switched on. But these files were produced by a tool that converted all svg files in a folder at once. So I probably have to go through all files and safe them once.

File: app/build.gradle (hunk 1 above)
Comment (biafra23) [reply app/build.gradle:118]: Looks as if this whole thing isn't even necessary. The font is picked up without this.

**Comment by biafra23** [comment]
> Separators for items in Help Center are missing

Not sure this is within the scope of this ticket as we didn't have a separator between SettingsItems. We used the background color as a separator.

**Comment by biafra23** [comment]
I tried to fix Help Center and Appearance. Please have a look!

**Comment by elgatovital** [comment]
help center and appearance look nice now 👍 Everything looks good except for following issues:
- There are some redundant lines in settings
Screenshot 2022-09-29 at 15 53 15
- Tx list is missing separators in dark mode
- Collectibles list is missing separators in dark mode

**Comment by elgatovital** [comment]
one critical issue is that collectibles images and placeholders are not displayed in dark mode
image

**Comment by TanyaEfremova** [comment]
Body texts should use labelSecondary to distinguish it from the headings and links: 
image

Here is how it looks in [design](https://www.figma.com/file/xxCMH9tgwOpaQZ4Fzhjnf2/Master-File?node-id=121%3A21829)

**Comment by biafra23** [comment]
@TanyaEfremova 
The text above the link is label_secondary. Which is `slate` in light mode and `stone` in dark mode.

**Comment by biafra23** [comment]
@elgatovital I cannot reproduce the problem with collectibles. Do you have Görli safe to reproduce this? My ENS collectibles look fine.

File: app/src/main/java/io/gnosis/safe/ui/settings/safe/SafeSettingsFragment.kt [hunk 4]
```diff
@@ -114,51 +114,53 @@ class SafeSettingsFragment : BaseViewBindingFragment<FragmentSettingsSafeBinding
             localName.name = safe?.localName
             threshold.name = getString(R.string.safe_settings_confirmations_required, safeInfo?.threshold, safeInfo?.owners?.size)
             ownersContainer.removeAllViews()
-            safeInfo?.owners?.forEach { owner -> ownersContainer.addView(ownerView(safe!!.chain, owner, localOwners)) }
+            safeInfo?.owners?.dropLast(1)?.forEach { owner -> ownersContainer.addView(ownerView(safe!!.chain, owner, localOwners)) }
+            ownersContainer.addView(safeInfo?.owners?.last()?.let { ownerView(safe!!.chain, it, localOwners, false) })
+
             masterCopy.setAddress(safe?.chain, safeInfo?.implementation?.value, safeInfo?.version)
             masterCopy.loadKnownAddressLogo(safeInfo?.implementation?.logoUri, safeInfo?.implementation?.value)
+            //masterCopy.showSeparator = false
```
Comment (elgatovital) [inline app/src/main/java/io/gnosis/safe/ui/settings/safe/SafeSettingsFragment.kt:122]: is this needed?

File: app/src/main/java/io/gnosis/safe/ui/settings/safe/SafeSettingsFragment.kt [hunk 5]
```diff
@@ -114,51 +114,53 @@ class SafeSettingsFragment : BaseViewBindingFragment<FragmentSettingsSafeBinding
             localName.name = safe?.localName
             threshold.name = getString(R.string.safe_settings_confirmations_required, safeInfo?.threshold, safeInfo?.owners?.size)
             ownersContainer.removeAllViews()
-            safeInfo?.owners?.forEach { owner -> ownersContainer.addView(ownerView(safe!!.chain, owner, localOwners)) }
+            safeInfo?.owners?.dropLast(1)?.forEach { owner -> ownersContainer.addView(ownerView(safe!!.chain, owner, localOwners)) }
```
Comment (elgatovital) [inline app/src/main/java/io/gnosis/safe/ui/settings/safe/SafeSettingsFragment.kt:117]: why is this needed? Sequence seems to be same as before, with the previous version using one forEach loop.

File: app/src/main/java/io/gnosis/safe/ui/settings/safe/SafeSettingsFragment.kt [hunk 6]
```diff
@@ -114,51 +114,53 @@ class SafeSettingsFragment : BaseViewBindingFragment<FragmentSettingsSafeBinding
...
+                background = ContextCompat.getDrawable(requireContext(), R.drawable.background_secondary_selectable)
                 layoutParams = LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, resources.getDimension(R.dimen.item_address).toInt())
                 setAddress(chain, owner.value, localOwner.type)
                 name = if (localOwner.name.isNullOrBlank())
                     context.getString(
                         R.string.settings_app_imported_owner_key_default_name,
                         localOwner.address.shortChecksumString()
                     ) else localOwner.name
-                showSeparator = true
+                if (separate) showSeparator = true
```
Comment (elgatovital) [inline app/src/main/java/io/gnosis/safe/ui/settings/safe/SafeSettingsFragment.kt:143]: showSeparator = separate?

File: app/src/main/java/io/gnosis/safe/ui/settings/safe/SafeSettingsFragment.kt (hunk 4 above)
Comment (biafra23) [reply app/src/main/java/io/gnosis/safe/ui/settings/safe/SafeSettingsFragment.kt:122]: No. I will delete it.

File: app/src/main/java/io/gnosis/safe/ui/settings/safe/SafeSettingsFragment.kt (hunk 5 above)
Comment (biafra23) [reply app/src/main/java/io/gnosis/safe/ui/settings/safe/SafeSettingsFragment.kt:117]: I found something more readable without the index. 

```
safeInfo?.owners?.forEach() { owner ->
    ownersContainer.addView(ownerView(safe!!.chain, owner, localOwners, owner != safeInfo.owners.last()))
}
```

**Review by elgatovital — APPROVED** [review]
